src/aics-image-viewer/components/AxisClipSliders/index.tsx

```diff
@@ -71,183 +88,184 @@ const SliderRow: React.FC<SliderRowProps> = ({ label, vals, valsReadout = vals,
+    pauseOnInput(axis);
 
-  updateRegion(axis: AxisName, minval: number, maxval: number): void {
-    const { changeViewerSetting, numSlices, region } = this.props;
+    const { changeViewerSetting, numSlices, region } = props;
     // get a value from 0-1
     const max = numSlices[axis];
     const start = minval / max;
     const end = maxval / max;
     changeViewerSetting("region", { ...region, [axis]: [start, end] });
-  }
-
-  updateSlice(axis: AxisName, val: number): void {
-    const { changeViewerSetting, numSlices, slices } = this.props;
-    changeViewerSetting("slice", { ...slices, [axis]: val / numSlices[axis] });
-  }
-
-  makeSliderCallback(axis: AxisName): (values: number[]) => void {
-    return (values: number[]) => {
-      if (values.length < 2) {
-        this.updateSlice(axis, values[0]);
-      } else {
-        this.updateRegion(axis, values[0], values[1]);
-      }
-    };
-  }
-
-  createAxisSlider(axis: AxisName, twoD: boolean): React.ReactNode {
-    const { playing } = this.state;
-    const numSlices = this.props.numSlices[axis];
-    const clipVals = this.props.region[axis];
-    const slice = this.props.slices[axis];
-    const sliderVals = twoD
-      ? [Math.round(slice * numSlices)]
-      : [Math.round(clipVals[0] * numSlices), Math.round(clipVals[1] * numSlices)];
+  };
+
+  const updateSlice = (axis: AxisName, slice: number): void => {
+    pauseOnInput(axis);
+    props.changeViewerSetting("slice", { ...props.slices, [axis]: slice / props.numSlices[axis] });
+  };
+
+  const updateTime = (time: number): void => {
+    pauseOnInput("t");
+    props.changeViewerSetting("time", time);
+  };
+
+  // Pause when view mode or volume size has changed
+  useEffect(() => props.playControls.pause(), [props.mode, ...Object.values(props.numSlices)]);
+
+  const handlePlayPause = (axis: AxisName | "t", willPlay: boolean): void => {
+    if (willPlay) {
+      props.playControls.play(axis);
+    } else {
+      props.playControls.pause();
+    }
+  };
+
+  const create2dAxisSlider = (axis: AxisName): React.ReactNode => {
+    const numSlices = props.numSlices[axis];
+    const numSlicesLoaded = props.numSlicesLoaded[axis];
 
     return (
-      <div key={axis + numSlices} className={`slider-row slider-${axis}`}>
-        <SliderRow
-          // prevents slider from potentially not updating number of handles
-          key={`${twoD}`}
+      <div key={axis + numSlices + numSlicesLoaded} className={`slider-row slider-${axis}`}>
+        <PlaySliderRow
           label={axis.toUpperCase()}
-          vals={sliderVals}
-          max={numSlices - (twoD && numSlices > 1 ? 1 : 0)}
-          onSlide={this.makeSliderCallback(axis)}
+          val={Math.round(props.slices[axis] * numSlices)}
+          max={numSlices}
+          onChange={(val) => updateSlice(axis, val)}
+          onStart={() => props.playControls.startHold(axis)}
+          onEnd={() => props.playControls.endHold()}
```

Comment:
I feel like `onStart` and `onEnd` on their own aren't super descriptive... maybe `onStartInteraction` would be better?

Reply:
These are mainly named for parity with the noUiSlider events they eventually attach to: https://refreshless.com/nouislider/events-callbacks/

Reply:
a one line comment to say "these names map with nouislider event names" is a good idea.